### PR TITLE
exporting volume (volr) to land model and handling negative qgwl

### LINKED ISSUE #164: Handle qgwl like MOSART does
Currently we are ignoring the QGWL Glacier/Wetland/Lake residual term from the coupler. This should be handled similar to how it's handled in MOSART. This likely means there should be several different options on how to handle it.

### PR DESCRIPTION
**Pull request by nmizukami**
- turn on volume export
- handling negative qgwl.  Three options: putting 1) all, 2) negative, or 3) > threshold qgwl to `rtmCTL%direct` array 

resolve #164

### CODE REVIEW

File: route/build/cpl/RtmVar.F90 [hunk 1]
```diff
@@ -46,14 +46,17 @@ MODULE RtmVar
...
-  integer,           public            :: coupling_period                  ! coupling period
-  integer,           public            :: rtmhist_ndens  = 1               ! namelist: output density of netcdf history files
-  integer,           public            :: rtmhist_mfilt  = 30              ! namelist: number of time samples per tape
-  integer,           public            :: rtmhist_nhtfrq = 0               ! namelist: history write freq(0=monthly)
-  logical,           public            :: ice_runoff     = .false.         ! true => runoff is split into liquid and ice,
+  logical,           public            :: do_rof         = .true.                   !
+  logical,           public            :: do_flood       = .false.                  !
+  character(len=CL), public            :: nrevsn_rtm     = ' '                      ! restart data file name for branch run
+  character(len=32), public            :: qgwl_runoff_option ='threshold'           ! method for handling qgwl runoff
+  character(len=32), public            :: bypass_routing_option = 'direct_in_place' ! bypass routing model method
```
[the remark this answers is not in the document]
Comment (ekluzek) [reply route/build/cpl/RtmVar.F90:53]: Looks like currently it's hardcoded to only allow direct_in_place

**Review by ekluzek — CHANGES_REQUESTED** [review]
The new control variables need to be added to the settings control file or the namelist. It also looks like they aren't fully enabled yet. I assume you will fully enable them in this PR?

File: route/build/cpl/RtmMod.F90 [hunk 2]
```diff
@@ -480,14 +566,24 @@ SUBROUTINE get_river_export_data(NETOPO_in, RCHFLX_in)
...
           nCatch = size(NETOPO_in(iRch)%HRUIX)
           do iHru = 1, nCatch
             ix = NETOPO_in(iRch)%HRUIX(iHru)
-            rtmCTL%volr(ix)        = RCHFLX_in(iens, iRch)%ROUTE(idxIRF)%REACH_VOL(1)
-            rtmCTL%flood(ix)       = 0._r8
+
+            ! stream volume is distributed into HRUs based on HRU's areal weight for contributory area
+            vol_hru = RCHFLX_in(iens,iRch)%ROUTE(idxIRF)%REACH_VOL(1) * NETOPO_in(iRch)%HRUWGT(iHru)
+            if (vol_hru > 0._r8) then
+              rtmCTL%volr(ix) = vol_hru/rtmCTL%area(ix)
```
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/cpl/RtmMod.F90:582]: without this if statement, volr get NaN at headwater HRUs (I believe) where volume is zero. Will investigate this later.

File: route/build/cpl/RtmVar.F90 [hunk 3]
```diff
@@ -46,14 +46,17 @@ MODULE RtmVar
...
-  character(len=CL), public            :: nrevsn_rtm     = ' '             ! restart data file name for branch run
-  integer,           public            :: coupling_period                  ! coupling period
-  integer,           public            :: rtmhist_ndens  = 1               ! namelist: output density of netcdf history files
-  integer,           public            :: rtmhist_mfilt  = 30              ! namelist: number of time samples per tape
-  integer,           public            :: rtmhist_nhtfrq = 0               ! namelist: history write freq(0=monthly)
-  logical,           public            :: ice_runoff     = .false.         ! true => runoff is split into liquid and ice,
+  logical,           public            :: do_rof         = .true.                   !
+  logical,           public            :: do_flood       = .false.                  !
+  character(len=CL), public            :: nrevsn_rtm     = ' '                      ! restart data file name for branch run
+  character(len=32), public            :: qgwl_runoff_option ='threshold'           ! method for handling qgwl runoff
```
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/cpl/RtmVar.F90]: qgwl_runoff_option is moved to saved variables in mizuRoute side

File: route/build/cpl/RtmVar.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/cpl/RtmVar.F90]: `bypass_routing_option` is moved to saved variables in mizuRoute shared code and both `qgwl_runoff_option` and `bypass_routing_option` are provided in control file

**Comment by ekluzek** [comment]
@nmizukami and I looked at this together and this is ready to bring in.

**Pull request merged**